lamindb/_record.py

```diff
@@ -847,6 +864,32 @@ def check_name_change(record: Record):
                 raise RecordNameChangeIntegrityError
 
 
+def check_key_change(record: Record):
+    """Errors if a record's key has falsely changed."""
+    if not record.pk or record.key is None or not hasattr(record, "_key"):
```

Comment:
Can you add comments on these conditions?

I don't understand the first two. The last two is moot if you have proper type checks for Transform and Artifact; please refactor.

If record.key is None and people set it to something invalid, that's also a problem; so I don't think this should return here, should it?

Reply:
If key was `None` people can set it, yes. It has then to be set for all members of the version family.